Configurations/50-win-onecore.conf

```diff
@@ -36,13 +36,14 @@ my %targets = (
         # /NODEFAULTLIB:kernel32.lib is needed, because MSVCRT.LIB has
         # hidden reference to kernel32.lib, but we don't actually want
         # it in "onecore" build.
-        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        # /APPCONTAINER is needed for Universal Windows Platform compat
+        lflags          => add("/NODEFAULTLIB:kernel32.lib /APPCONTAINER"),
         defines         => add("OPENSSL_SYS_WIN_CORE"),
         ex_libs         => "onecore.lib",
     },
     "VC-WIN64A-ONECORE" => {
         inherit_from    => [ "VC-WIN64A" ],
-        lflags          => add("/NODEFAULTLIB:kernel32.lib"),
+        lflags          => add("/NODEFAULTLIB:kernel32.lib /APPCONTAINER"),
```

Comment:
ONECORE is a single entrypoint to use Win32 APIs that are available on all Windows 10 targets (however this isn't strictly true because some disallowed APIs will only spring errors when you run WACK), and also on older versions of Windows (subject to the API minimum versions of each function / symbol).

The Universal Windows Platform (UWP) is ONECORE Win32 APIs + [WinRT](https://en.wikipedia.org/wiki/Windows_Runtime) which is an entirely new set of APIs, available via C++, C#, JS.

As far as I know, you'd target ONECORE to build a binary that uses Win32 APIs and works on UWP. People building win32 apps would use the older windows configure targets.

However, even if there were other reasons, building with `/APPCONTAINER` should not change anything for openssl. These are the restrictions it applies: https://docs.microsoft.com/en-us/windows/win32/secauthz/appcontainer-isolation. Overall it actually improves openssl security.